### PR TITLE
Add Gradle sub-projects to enable parallel running of Kafka tests (#2…

### PR DESCRIPTION
**Pull request by johnjcasey**
…6153)

* add parallel number to gradle.properties

* remove unused config, attempt to configure higher parallelism for kafka tests

* Create generic task to represent Integration tests. Then, create sub-projects to enable running tests in parallel

* extend test timeout to reduce flakyness

* add comment for running locally

* run spotless

* factor kafka integration tests up a level

(cherry picked from commit 837733e862007cbeba0a57f682109d9debec9340)

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Review by jrmccluskey — APPROVED** [review]

File: buildSrc/src/main/groovy/org/apache/beam/gradle/kafka/KafkaTestUtilities.groovy [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import org.gradle.api.Project
+import org.gradle.api.artifacts.ConfigurationContainer
+import org.gradle.api.tasks.testing.Test
+
+import javax.inject.Inject
+
+class KafkaTestUtilities {
+  abstract static class KafkaBatchIT extends Test {
+
+    @Inject
```
Comment (kennknowles) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/kafka/KafkaTestUtilities.groovy:29]: I'm not familiar with this. Can you document the control/data flow? I'm assuming the idea is that we get default values for most things and can tweak them via passing parameters in a particular way.

**Comment by Abacn** [comment]
stop reviewer notifications

**Comment by jrmccluskey** [comment]
Run Java PreCommit

**Comment by jrmccluskey** [comment]
Run Java_PVR_Flink_Batch PreCommit

**Comment by jrmccluskey** [comment]
Run Python PreCommit

**Comment by jrmccluskey** [comment]
Run Python_Coverage PreCommit

**Comment by jrmccluskey** [comment]
Run Python_Integration PreCommit

**Comment by jrmccluskey** [comment]
Run Python_Integration PreCommit

**Comment by jrmccluskey** [comment]
Run Java PreCommit

**Comment by jrmccluskey** [comment]
Run Java_PVR_Flink_Batch PreCommit

**Comment by jrmccluskey** [comment]
Run Python PreCommit

File: settings.gradle.kts [hunk 2]
```diff
@@ -284,3 +284,21 @@ project(":beam-test-jenkins").projectDir = file(".test-infra/jenkins")
 include("beam-validate-runner")
 project(":beam-validate-runner").projectDir = file(".test-infra/validate-runner")
 include("com.google.api.gax.batching")
+include("sdks:java:io:kafka:kafka-251")
+findProject(":sdks:java:io:kafka:kafka-251")?.name = "kafka-251"
```
Comment (kennknowles) [inline settings.gradle.kts:288]: what is this about? comment in file please

**Comment by kennknowles** [comment]
This was merged with unresolved comment threads that I really would like clarified. This introduces a number of potentially esoteric new styles to the codebase that require some signposting.

**Comment by Abacn** [comment]
This is a cherry pick PR to release-2.47.0 branch so it has to keep the commit the same as master. It is already in master (#26153) the comments should be resolved as a forward fix on master CC: @johnjcasey

**Comment by jrmccluskey** [comment]
+1

**Pull request merged**